base/shell/cmd/cmd.c

```diff
@@ -2299,11 +2345,20 @@ Initialize(VOID)
     if (!*ptr)
     {
         /* If neither /C or /K was given, display a simple version string */
-        ConOutChar(_T('\n'));
+
+        /* Print the information bar if its environment variable is set. */
```

Comment:
Please revert these changes. If the info line is printed when the prompt is shown, this is because when the console scrolls past one screenful, the info line would be lost if it is printed only once as you do it there now.

Reply:
The new line above the copyright notice is a hack.

Reply:
> the top of the window after the prompt, but rather at the top of the entire console history

It is redrawn at the top of the **console screenbuffer** as it should be. (I tested on OS/2 etc.)

Reply:
Note that when I said "these changes" I was talking about moving the printInfoLine function to cmd.c and removing the /I prompt switch.

Reply:
Well if that's how it's supposed to work, guess that's fine. Just seemed odd to me. 

I can move the line draw code back into `prompt.c` and write a function in `cmd.c` to determine if the prompt environment variable has $I in it. Then insert a new line above the copyright notice only if it returns true.

In the future, the information line would be a lot more useful if it had its own environment variable to customize it.

Reply:
> I can move the line draw code back into prompt.c and write a function to determine if the prompt environment variable has $I in it. Then insert a new line above the copyright notice only if it returns true.

Yes. That code: https://github.com/reactos/reactos/pull/5745/files#diff-499121b37bcb4ad5b9aad691293f228a3afe7d2de8e81960459033a04c9d8a21L120
(that you removed in the current state of your PR as I'm currently writing this comment) could be made into a helper function, like `BOOL IsInfoLineShown(VOID)` (or similar name), that you could call in where the extra `\n` is issued in the Initialize() function.
(And that helper also called where the original removed code was, if it still makes sense.)

> In the future, the information line would be a lot more useful if it had its own environment variable to customize it.

Still possible. Where there is this call in the PrintInfoLine function:
```c
    iInfoLineLen = LoadString(CMD_ModuleHandle, STRING_CMD_INFOLINE, (PTSTR)&pszInfoLine, 0);
```
you could do, schematically,
```c
...
    TCHAR szInfoBar[256];
...
    iInfoLineLen = GetEnvironmentVariable(_T("INFOLINE"), szInfoBar, _countof(szInfoBar));
    if (iInfoLineLen != 0 && *szInfoBar) // You should check whether *szInfoBar check is necessary in that case...
    {
        pszInfoLine = szInfoBar;
    }
    else
    {
        iInfoLineLen = LoadString(CMD_ModuleHandle, STRING_CMD_INFOLINE, (PTSTR)&pszInfoLine, 0);
    }
...
```
Also I'm not sure about the magic "0" and "1" strings you are currently using in your PR.

Reply:
I moved the draw info line code back to `prompt.c` and added a new function to check if the info line is displayed or not.